### PR TITLE
Doc: PRC Layer core

### PR DESCRIPTION
**Pull request by VenkyOS**
Working on correcting images. Content is fine.

### CODE REVIEW

File: doc/RPC_Layer_Core.rst [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+RPC Layer Core
+==================
+
+This document presents a high level design (HLD) of rpc layer core of Colibri C2 core. The main purposes of this document are: (i) to be inspected by C2 architects and peer designers to ascertain that high level design is aligned with C2 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document. The intended audience of this document consists of C2 customers, architects, designers and developers.
+
+**************
+Introduction
+**************
+
+RPC layer is a high level part of C2 network stack. It contains operations allowing user to send RPC items, containing File Operation Packets (FOP) to network destinations. RPC layer speaks about FOPs being sent from and to end-points and RPCs, which are containers for FOPs, being sent to services. RPC layer is a part of a C2 server as well as a C2 client.
```
Comment (huanghua78) [inline doc/RPC_Layer_Core.rst:11]: Rename all "C2" to Motr.

File: doc/RPC_Layer_Core.rst [hunk 2]
```diff
@@ -0,0 +1,266 @@
+==================
+RPC Layer Core
+==================
+
+This document presents a high level design (HLD) of rpc layer core of Colibri C2 core. The main purposes of this document are: (i) to be inspected by C2 architects and peer designers to ascertain that high level design is aligned with C2 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document. The intended audience of this document consists of C2 customers, architects, designers and developers.
```
Comment (huanghua78) [inline doc/RPC_Layer_Core.rst:5]: Rename all Colibri to "CORTX Motr"

File: doc/RPC_Layer_Core.rst [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+Introduction
+**************
+
+RPC layer is a high level part of C2 network stack. It contains operations allowing user to send RPC items, containing File Operation Packets (FOP) to network destinations. RPC layer speaks about FOPs being sent from and to end-points and RPCs, which are containers for FOPs, being sent to services. RPC layer is a part of a C2 server as well as a C2 client.
+
+**************
+Definitions
+**************
+
+- fop, file operation packet, a description of file operation suitable for sending over network and storing on a storage device. File operation packet (FOP) identifies file operation type and operation parameters;
```
Comment (huanghua78) [inline doc/RPC_Layer_Core.rst:17]: use the following to make the font BOLD:
```
*fop*
```

File: doc/RPC_Layer_Core.rst [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+  - By the endpoint item (FOP, ADDB) is following to (sessions);
+
+  - By priority (assigned via ext. interface).
+
+- Formation subcomponent proceeds with classification by some criteria and produces RPCs which are containers of RPC items. It is responsible for data preparation in an order that is optimal for network utilization. The following options are used:
+
+  - Do grouping by fid of file;
+
+  - Maximal count of RPCs in processing (C2_MAX_RPCS_IN_FLIGHT);
```
Comment (huanghua78) [inline doc/RPC_Layer_Core.rst:98]: Rename all prefix "C2_" to "M0_"

File: doc/RPC_Layer_Core.rst [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+- void c2_rpc_session_unlink(c2_rpc_session_id);
+
+- void c2_rpc_slot_unlink(c2_rpc_slot_id);
+
+- void c2_rpc_slot_max_set(uint32_t max);
+
+Processing IF:
+
+- int c2_rpc_item_send(c2_rpc_item*, c2_service_id, c2_rpc_prio, c2_rpc_caching_type, c2_rpc_callback*);
```
Comment (huanghua78) [inline doc/RPC_Layer_Core.rst:196]: rename all "c2_" prefix to "m0_" prefix.

File: doc/RPC_Layer_Core.rst (hunk 2 above)
Comment (VenkyOS) [reply doc/RPC_Layer_Core.rst]: Incorporated.

File: doc/RPC_Layer_Core.rst (hunk 3 above)
Comment (VenkyOS) [reply doc/RPC_Layer_Core.rst]: Incorporated.

File: doc/RPC_Layer_Core.rst (hunk 5 above)
Comment (VenkyOS) [reply doc/RPC_Layer_Core.rst]: Incorporated.

File: doc/RPC_Layer_Core.rst (hunk 4 above)
Comment (VenkyOS) [reply doc/RPC_Layer_Core.rst]: Incorporated.

File: doc/RPC_Layer_Core.rst (hunk 1 above)
Comment (VenkyOS) [reply doc/RPC_Layer_Core.rst]: Incorporated.

**Review by mukul-seagate11 — APPROVED** [review]
looks good

**Review by huanghua78 — CHANGES_REQUESTED** [review]
Compared with the original doc, there are some figures dropped. Why?

Also the references are dropped.

File: doc/RPC_Layer_Core.rst (hunk 3 above)
Comment (huanghua78) [reply doc/RPC_Layer_Core.rst]: I mean, to make all the definition keywords in this section BOLD, not only the `fop`.

**Comment by VenkyOS** [comment]
> Compared with the original doc, there are some figures dropped. Why?
> 
> Also the references are dropped.

I am making the necessary corrections.

**Comment by huanghua78** [comment]
Is this ready?
All the images added?

**Comment by VenkyOS** [comment]
> Is this ready?
> All the images added?

Not yet. Two more images are to be added.

**Comment by huanghua78** [comment]
Hello @VenkyOS ,
I think there are two more images pending for this PR.
What is the status?

**Comment by VenkyOS** [comment]
> Hello @VenkyOS ,
> I think there are two more images pending for this PR.
> What is the status?

@huanghua78 , I am working on this. Will update you soon.

**Comment by huanghua78** [comment]
@VenkyOS , please complete this PR when you have time.

**Comment by VenkyOS** [comment]
> @VenkyOS , please complete this PR when you have time.

The images have been updated.

**Review by mukul-seagate11 — APPROVED** [review]
looks good

**Review by huanghua78 — APPROVED** [review]

**Pull request merged**